Code/Source/Lidar/PipelineGraph.h

```diff
@@ -0,0 +1,100 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <rgl/api/core.h>
+
+namespace RGL
+{
+    //! Class that manages the RGL pipeline graph construction, which depends on
+    //! three conditions: point-cloud compact, noise and publication. Below is a
+    //! visual representation of the pipeline graph.
+    //!
+    //!      +--------+    +--------------+    +-?--?--?--?-+
+    //!      |rayPoses+---->lidarTransform+---->angularNoise?
+    //!      +--------+    +--------------+    +-?--?--?--?++
+    //!                                                    |
+    //!      +-?--?---?--?-+     +-?--?---?--?-+      +----v---+
+    //!      |pointsCompact<-----+distanceNoise<------+rayTrace|
+    //!      +-?--?-+-?--?-+     +-?--?-?-?--?-+      +--------+
+    //!             |                   |
+    //!      +------v-----+      +------v--------+
+    //!      |pointsFormat|      |pointsTransform|
+    //!      +------------+      +------+--------+
+    //!                                 |
+    //!                        +--------v----------+
+    //!                        |pointsFormatPublish|
+    //!                        +--------+----------+
+    //!                                 |
+    //!                         +-------v---------+
+    //!                         |pointCloudPublish|
+    //!                         +-----------------+
+    //!
+    //! Enabling compact adds an additional pointsCompact node to the pipeline,
+    //! enabling noise, adds the angularNoise and distance noise nodes whereas
+    //! enabling point-cloud publishing adds an additional branch to the pipeline
+    //! graph consisting of three nodes: pointsTransform, pointsFormatPublish
+    //! and pointCloudPublish.
+    class PipelineGraph
+    {
+    public:
+        struct Nodes
+        {
+            rgl_node_t m_rayPoses{ nullptr }, m_lidarTransform{ nullptr }, m_angularNoise{ nullptr }, m_rayTrace{ nullptr },
+                m_distanceNoise{ nullptr }, m_pointsCompact{ nullptr }, m_pointsFormat{ nullptr }, m_pointCloudTransform{ nullptr },
+                m_pointsFormatPublish{ nullptr }, m_pointCloudPublish{ nullptr };
+        };
+
+        PipelineGraph(float maxRange, AZStd::vector<rgl_field_t>& resultFields);
+        PipelineGraph(const PipelineGraph& other) = delete;
+        PipelineGraph(PipelineGraph&& other);
+        ~PipelineGraph();
+
+        [[nodiscard]] bool IsCompactEnabled() const
+        {
+            return m_isCompactEnabled;
+        };
+        [[nodiscard]] bool IsPublishingEnabled() const
+        {
+            return m_isPublishingEnabled;
+        };
+        [[nodiscard]] bool IsNoiseEnabled() const
+        {
+            return m_isNoiseEnabled;
+        }
+        
+        [[nodiscard]] bool IsPublisherConfigured() const
+        {
+            return m_nodes.m_pointCloudPublish;
```

Comment:
```suggestion
            return m_nodes.m_pointCloudPublish == nullptr;
```
For clarity.

Reply:
Implicit casting pointers to booleans has been suggested to me before as a popular practice in the O3DE's code.